### DISCUSSION TITLE
[Console] Simplify Creation of Interpolated Environment Variables for Microservices

### DISCUSSION THREAD

**Discussion by GioAc96** [mia-platform/community#213] [Ideas]
HI! When adding environment variables to microservices I often need to create variables with the same name as other public/env variables which they interpolate to (eg: I often need to create a variable called `MONGO_URL` with its value set to `{{MONGO_URL}}`). Moreover, I often find myself creating several environment variables at once, which makes the process more error-prone. Simplifying the creation of such variables would improve the developer experience by:

- minimizing the probability of mistakes or typos when interpolating variables
- reducing the cases of accidental incoherent variable naming
- speeding up the the configuration of microservices

One way this could be achieved would be to create a separate flow for adding interpolated variables, maybe using a drop-down menu which allows to select among existing public/env variables.

**Comment by fredmaggiowski** [comment]
Hi @GioAc96 seems like an interesting feature; I'd suggest you to track it in the internal product backlog

@albertotessarotto @jacoposte FYI

**Reply by albertotessarotto** [reply]
@jacoposte we may include it among the various ux proposals concerning the improvement of the environment variable management part

**Reply by mattia-fumo** [reply]
If I recall correctly we already have a similar feature in out backlog, @jacoposte am I right?
